### ISSUE TITLE
WSO2 API-M 4.1.0: 404 problem with new API

### ISSUE THREAD

**Issue by engineering-dotcms** [wso2/api-manager#60]
### Description:

I was able to create and publish APIs on my dev WSO2 installation based on H2 DB. Unfortunately started from the fifth APIs (the first four still working very well) I have this error.

If I'm calling the API from Postman I receive

{ "code": 404, "type": "Status report", "message": "Not Found", "description": "The requested resource is not available." }

Into the logs I see

INFO {org.apache.synapse.mediators.builtin.LogMediator} - STATUS = Message dispatched to the main sequence. Invalid URL., RESOURCE = XXXXXXXXXXXXXXX, HEALTH CHECK URL = XXXXXXXXXXXXXXX

Any help will appriciate.

### Affected product version:
API-M version 4.1.0

### Affected component:
API-M version 4.1.0

### Environment details (with versions):
- OS: Ubuntu 18.04
- Client: Ubuntu 18.04
- Env (Docker/K8s): standalone installation with H2 DB

---

**Comment by dushaniw** [comment]
When this fifth API is deployed from the Publisher portal, do you see the "synpase config successfully deployed" info logs in the wso2carbon.log file?

Ex:
`[2022-05-09 12:08:31,802]  INFO - DependencyTracker API : PizzaShackAPI:v1.0.0 was added to the Synapse configuration successfully
`

**Comment by kyawswar87** [comment]
I am facing that issue too. But for me, there is no such issue on H2 db. But for mysql, I got such error. Unluckily I got such issue on the creation of first endpoint.

When I change mysql db, I have to restart the APM server on every time I created new endpoints. If I didn't restart APM server, I cannot call the created endpoints. I got 404 response.

**Comment by kyawswar87** [comment]
@dushaniw Yes. I can see these log if I am using H2 db. But I cannot see that log in using Mysql.

![image](https://user-images.githubusercontent.com/105942825/169663505-6c7e5861-7b4b-4f9b-9675-d7580a96cf23.png)

**Comment by imthy** [comment]
I'm using PostgreSQL with APIM 4.1.0 and facing the same issue. In my case, I'm also getting the message synapse config successfully deployed in the logs. But, when I try to invoke the api, am getting 404 error. This worked fine in my APIM 3.2.0 and the issue is only with 4.1.0

**Comment by engineering-dotcms** [comment]
Resolution/Invalid

**Comment by kyawswar87** [comment]
@engineering-dotcms May I know your finding? I think we have different issue. Thanks.

**Comment by marmo94** [comment]
@engineering-dotcms Could you please tell us what was the sollution for you. I'm facing the same issue with Postgres SQL.
Many thanks
